internal/controller/gateway.go

```diff
@@ -189,11 +196,150 @@ func (c *GatewayController) ensureExtensionPolicy(ctx context.Context, gw *gwapi
+		// No EPP service configured, skip.
+		return nil
+	}
+
+	eppServiceName := string(inferencePool.Spec.ExtensionRef.Name)
+	var eppServicePort int32 // Default port.
+	if inferencePool.Spec.ExtensionRef.PortNumber != nil {
+		eppServicePort = int32(*inferencePool.Spec.ExtensionRef.PortNumber)
+	} else {
+		// The port number on the service running the extension. When unspecified,
+		// implementations SHOULD infer a default value of 9002 when the Kind is
+		// Service.
+		eppServicePort = 9002
+	}
+
+	// Create Backend resource for the EPP service.
+	backendName := fmt.Sprintf("epp-%s-%s", route.Name, strings.ReplaceAll(inferencePoolKey, "/", "-"))
+	backend := egv1a1.Backend{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backendName,
+			Namespace: gw.Namespace,
+		},
+		Spec: egv1a1.BackendSpec{
+			Endpoints: []egv1a1.BackendEndpoint{{
+				FQDN: &egv1a1.FQDNEndpoint{
+					Hostname: fmt.Sprintf("%s.%s.svc.cluster.local", eppServiceName, route.Namespace),
+					Port:     eppServicePort,
+				},
+			}},
+		},
+	}
+
+	if err := c.client.Create(ctx, &backend); err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create EPP backend: %w", err)
+	}
+
+	// Create EnvoyExtensionPolicy for the EPP service, targeting the HTTPRoute.
+	eppPolicyName := fmt.Sprintf("epp-%s-%s", route.Name, strings.ReplaceAll(inferencePoolKey, "/", "-"))
+	eppExtPolicy := &egv1a1.EnvoyExtensionPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eppPolicyName,
+			Namespace: gw.Namespace,
+		},
+		Spec: egv1a1.EnvoyExtensionPolicySpec{
+			PolicyTargetReferences: egv1a1.PolicyTargetReferences{TargetRefs: []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName{
+				{
+					LocalPolicyTargetReference: gwapiv1a2.LocalPolicyTargetReference{
+						Kind:  "HTTPRoute",
+						Group: "gateway.networking.k8s.io",
+						Name:  gwapiv1.ObjectName(route.Name),
+					},
+				},
+			}},
+			ExtProc: []egv1a1.ExtProc{{
+				ProcessingMode: &egv1a1.ExtProcProcessingMode{
+					AllowModeOverride: true,
+					// For EPP, we only need headers (no body processing).
+					// When Request/Response fields are present, headers are automatically sent.
+					Request: &egv1a1.ProcessingModeOptions{
+						Body: ptr.To(egv1a1.BufferedExtProcBodyProcessingMode),
+					}, // Headers sent, no body.
+					Response: &egv1a1.ProcessingModeOptions{
+						Body: ptr.To(egv1a1.BufferedExtProcBodyProcessingMode),
+					}, // Headers sent, no body.
+				},
+				BackendCluster: egv1a1.BackendCluster{
+					BackendRefs: []egv1a1.BackendRef{{
+						BackendObjectReference: gwapiv1.BackendObjectReference{
+							Name:      gwapiv1.ObjectName(backendName),
+							Kind:      ptr.To(gwapiv1.Kind("Backend")),
+							Group:     ptr.To(gwapiv1.Group("gateway.envoyproxy.io")),
+							Namespace: ptr.To(gwapiv1.Namespace(gw.Namespace)),
+						},
+					}},
+				},
+			}},
+		},
+	}
+
+	if err := c.client.Create(ctx, eppExtPolicy); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
Same here need to handle the update case if the resource exists already, we ran into issue before if the extension policy already exists then it won't update when ai gateway controller code changes here like updating processing mode etc.

Reply:
instead of creating eep, how about adding extproc into the specific route which routes to the inference pool lb policy cluster? that why other normal routes won't need to talk to eep unnecessarily. This could be done in extension server i guess?